Automattic/jetpack
# build: Produce plugins artifact holding zips for all plugins

#### Changes proposed in this Pull Request:

This is the first step towards allowing the Jetpack Beta Tester plugin
to handle other plugins besides just Jetpack.
    
The build will still also produce the jetpack-dev artifact, so as to
continue working with the existing Beta Download website. That will be
removed in a future PR, after the Beta Download website has been updated
to download the new artifact instead.

This also moves Jetpack's `to-test.md` from the monorepo root into Jetpack,
so that other plugins can add their own `to-test.md` in the future.

#### Jetpack product discussion


p9dueE-35m-p2

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






* Check that the `jetpack-dev` artifact is still built, and has the same contents as before.
* Check that the `plugins` artifact is built, and contains a `jetpack-dev.zip` which contains contents equivalent to the `jetpack-dev` artifact.
* Create a PR based on this that adds `.extra.wp-plugin-slug` to another plugin, e.g. Backup. See that the `plugins` artifact now also contains an appropriately-named zip file for that plugin with a matching directory structure holding a built copy of that plugin.
  * Or look at https://github.com/Automattic/jetpack/actions/runs/957806774 if the artifact still exists.
* Check that https://betadownload.jetpack.me/jetpack-branches.json has an entry for this branch, with the appropriate version and jetpack-dev.zip linked.
